core/src/main/java/org/bitcoinj/net/discovery/DnsDiscovery.java

```diff
@@ -56,7 +56,7 @@ public DnsDiscovery(String[] dnsSeeds, NetworkParameters params) {
     }
 
     private static List<PeerDiscovery> buildDiscoveries(NetworkParameters params, String[] seeds) {
-        List<PeerDiscovery> discoveries = new ArrayList<PeerDiscovery>(seeds.length);
+        List<PeerDiscovery> discoveries = new ArrayList<PeerDiscovery>(seeds != null ? seeds.length : 0);
```

Comment:
This will just move the NPE one line lower.

Why not remove the entire parameter in the ArrayList constructor, it isn't much useful?